src/Features/Core/Portable/CodeCleanup/AbstractCodeCleanupService.cs

```diff
@@ -136,7 +156,49 @@ protected AbstractCodeCleanupService(ICodeFixService codeFixService)
             return document;
         }
 
+        private async Task<Document> ApplyThirdPartyCodeFixesAsync(Document document, IProgressTracker progressTracker, CodeActionOptions options, CancellationToken cancellationToken)
+        {
+            var tree = await document.GetRequiredSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
+            var range = new TextSpan(0, tree.Length);
+
+            // Compute diagnostics for everything that is not an IDE analyzer
+            var allDiagnostics = (await _diagnosticService.GetDiagnosticsForSpanAsync(document, range,
+                shouldIncludeDiagnostic: static diagnosticId => !(IDEDiagnosticIdToOptionMappingHelper.IsKnownIDEDiagnosticId(diagnosticId)),
+                includeSuppressedDiagnostics: false,
+                cancellationToken: cancellationToken).ConfigureAwait(false));
+
+            // ensure more than just compiler diagnostics were returned
+            var diagnostics = allDiagnostics.WhereAsArray(d => !IsCompilerDiagnostic(d.Id));
```

Comment:
Is there a reason not to just do this filtering in the `shouldIncludeDiagnostic` filter above?